hr_attendance_report_theoretical_time/models/hr_leave.py

```diff
@@ -1,12 +1,38 @@
 # Copyright 2017-2019 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import models
+from odoo import models, fields
 
 
 class HrLeave(models.Model):
     _inherit = 'hr.leave'
 
+    def _prepare_resource_leave_values(self):
+        return {
+            'date_from': fields.Datetime.to_string(
```

Comment:
Why do you need this?

Reply:
It's necessary because don't exist any function to prepare dict.
Is necessary to update resource.leave record fields: `date_from` or `date_to` (datetime fields) if leave had changed and our dates (hours).

Reply:
I think we are not thinking in the same things. Better to meet for joining efforts.